project/profiles/capacity4more/modules/c4m/og/c4m_og/c4m_og.module

```diff
@@ -161,10 +161,214 @@ function c4m_og_menu() {
+    'access callback' => 'c4m_og_control_membership_access',
+    'access arguments' => array(1, 2),
+  );
+
   return $items;
 }
 
 /**
+ * Function initializes form with input values, and returns confirm form.
+ *
+ * @param object $form
+ *   Form array to populate.
+ * @param object $form_state
+ *   Form state array.
+ * @param int $gid
+ *   Group ID.
+ * @param object $account
+ *   Administrator user that performs the action.
+ * @param object $membership
+ *   OG membership on which action is performed.
+ * @param int $timestamp
+ *   Timestamp of the moment action was requested.
+ * @param string $hashed_pass
+ *   Validation for executing user.
+ *
+ * @return object
+ *   Confirm reject form.
+ */
+function c4m_og_reject_membership_request($form, &$form_state, $gid, $account, $membership, $timestamp = 0, $hashed_pass = '') {
+  $form['reject_membership_request_message'] = array(
+    '#type' => 'textarea',
+  );
+
+  // Add group membership form.
+  $form_state['membership'] = $membership;
+  field_attach_form('membership', $membership, $form, $form_state);
+  $form_state['account'] = $account;
+  field_attach_form('account', $account, $form, $form_state);
+
+  $form['gid'] = array('#type' => 'value', '#value' => $gid);
+  $form['timestamp'] = array('#type' => 'value', '#value' => $timestamp);
+  $form['hashed_pass'] = array('#type' => 'value', '#value' => $hashed_pass);
+
+  return confirm_form($form, t('Please enter reject reason:'), 'group/node/' . $gid . '/admin/people', '', t('Reject'), t('Cancel'));
+}
+
+/**
+ * Submits reject membership request form.
+ *
+ * Gets inputs from form, and activates c4m_og_control_membership() function
+ * with action = 'reject'.
+ *
+ * @param object $form
+ *   Form array to populate.
+ * @param object $form_state
+ *   Form state array.
+ */
+function c4m_og_reject_membership_request_submit($form, &$form_state) {
+  $membership = $form_state['membership'];
+  $account = $form_state['account'];
+  $gid = $form_state['values']['gid'];
+  $timestamp = $form_state['values']['timestamp'];
+  $hashed_pass = $form_state['values']['hashed_pass'];
+
+  c4m_og_control_membership('reject', $gid, $account, $membership, $timestamp, $hashed_pass);
+}
+
+/**
+ * Function builds and returns a link for membership request control.
+ *
+ * Includes a token for auto login.
+ *
+ * @param object $account
+ *   Account which performs the action.
+ * @param object $membership
+ *   Membership for which action is performed.
+ * @param string $action
+ *   Approve / Reject.
+ *
+ * @return string
+ *   A link for membership request control.
+ */
+function c4m_og_autologin_control_membership_link($account, $membership, $action) {
+  $timestamp = REQUEST_TIME;
+  $options = array(
+    'absolute' => TRUE,
+    'purl' => array('disabled' => TRUE),
+  );
+
+  return url('group/node/' . $membership->gid . '/admin/membership-' . $action . '/' . $account->uid . '/'
```

Comment:
@anvmn 
The url seems complex anyway, please take it out to a variable so the `url()` will be cleaner, thanks.